app/controllers/pages_controller.rb

```diff
@@ -0,0 +1,69 @@
+require 'net/http'
+require 'json'
+
+POSTCODE_BASE_API = "https://api.postcodes.io/postcodes/"
+TFL_BASE_API = "https://api.tfl.gov.uk/StopPoint/"
+
+
+
+class PagesController < ApplicationController
+  def index
+  end
+
+  def postcode
+    lat,lon = get_coords_by_postcode
+    if lat == nil && lon == nil
+      flash[:error] = "Invalid postcode. Please try again."
+      redirect_to root_path
+    else
+      stop_points_information = get_stop_points_by_lat_lon(lat, lon)
+      if stop_points_information == nil
+        flash[:error] = "No bus stations nearby"
+        redirect_to root_path
```

Comment:
If moving error handling into request methods doesn't work might be worth putting this into a separate method, e.g. `get_stop_points_by_postcode`, that way the main method is only concerned with rendering the data